sycl/cmake/vecmem-setup-sycl.cmake

```diff
@@ -0,0 +1,45 @@
+# VecMem project, part of the ACTS project (R&D line)
+#
+# (c) 2024 CERN for the benefit of the ACTS project
+#
+# Mozilla Public License Version 2.0
+
+# VecMem include(s).
+include( vecmem-check-language )
+include( vecmem-check-sycl-source-compiles )
+
+# CMake include(s).
+include( CheckCXXSourceCompiles )
+
+# Helper function setting up the public definitions for vecmem::sycl
+#
+# Usage: vecmem_setup_sycl( vecmem::sycl )
+#
+function( vecmem_setup_sycl libName )
```

Comment:
Isn't it a bit strange to have a function (there's a similar one for the core lib) that accepts a library target but realistically is meant to be used with one single library that it applies to?

They get used in the installed config, but that doesn't require the library as an argument, does it?